producer-rest-core/src/main/scala/za/co/absa/spline/producer/rest/controller/ProducerStatusController.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.producer.rest.controller
+
+import io.swagger.annotations.ApiOperation
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation._
+import za.co.absa.spline.producer.service.repo.ExecutionProducerRepository
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@RestController
+class ProducerStatusController @Autowired()(
+  val repo: ExecutionProducerRepository) {
+
+  import ExecutionContext.Implicits.global
+
+  @GetMapping(value = Array("/status"))
+  @ApiOperation(
+    value = "/status",
+    notes =
+      """
+        Check that producer is running and that the database is accessible and initialized
+
+        ResponseBody format :
+
+        OK // when everything's working, or
+        FAIL // when there is a problem
+      """
+  )
+  def status(): Future[String] = {
+    repo.isDatabaseOk().map {
+      case true => "OK"
+      case false => "FAIL"
+    }
+  }
```

Comment:
Although technically it's fine, but why not doing it in a more RESTful way?
I would suggest using HTTP response status to communicate... well... the _status_ to the client, instead of encoding it into the response body. For example, we could use `200 OK` and `503 Service Unavailable` respectively.

Also, since RESTful gives a lot meaning to the HTTP methods, and we do not need to send any payload, I'd say we could find a better alternative to `GET` for this purpose.

[RFC-7231, § 4.3.2](https://tools.ietf.org/html/rfc7231#section-4.3.2)
> **HEAD**
> 
>    The HEAD method is identical to GET except that the server MUST NOT
>    send a message body in the response (i.e., the response terminates at
>    the end of the header section).  The server SHOULD send the same
>    header fields in response to a HEAD request as it would have sent if
>    the request had been a GET, except that the payload header fields
>    (Section 3.3) MAY be omitted.  This method can be used for obtaining
>    metadata about the selected representation without transferring the
>    representation data and **is often used for testing hypertext links for
>    validity, accessibility, and recent modification**.
> 
>    A payload within a HEAD request message has no defined semantics;
>    sending a payload body on a HEAD request might cause some existing
>    implementations to reject the request.
>